homeassistant/components/sensibo/climate.py

```diff
@@ -84,30 +78,42 @@
 HA_TO_SENSIBO = {value: key for key, value in SENSIBO_TO_HA.items()}
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType = None,
+):
     """Set up Sensibo devices."""
+    _LOGGER.warning(
+        "Loading Sensibo via platform setup is deprecated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            SENSIBO_DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Sensibo climate entry."""
+
     client = pysensibo.SensiboClient(
```

Comment:
Store the client in hass.data in init once you have passed all the potential retry points so you don't have to create it again